hardware/spectrometer/oceanoptics_spectrometer.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+"""
+This module controls spectrometers from Ocean Optics Inc.
+All spectrometers supported by python-seabreeze should work.
+
+Do "conda install -c poehlmann python-seabreeze"
+before using this module.
+
+Qudi is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Qudi. If not, see <http://www.gnu.org/licenses/>.
+
+Derived from the pyPL project (https://github.com/kaseyrussell/pyPL)
+Copyright 2010 Kasey Russell ( email: krussell _at_ post.harvard.edu )
+
+Copyright (c) the Qudi Developers. See the COPYRIGHT.txt file at the
+top-level directory of this distribution and at <https://github.com/Ulm-IQO/qudi/>
+
+"""
+
+from core.module import Base, ConfigOption
+from interface.spectrometer_interface import SpectrometerInterface
+import numpy as np
+import seabreeze.spectrometers as sb
+
+
+
+class OceanOptics(Base, SpectrometerInterface):
+    """ Hardware module for reading spectra from the WinSpec32 spectrometer software.
+
+    Example config for copy-paste:
+
+    myspectrometer:
+        module.Class: 'spectrometer.oceanoptics_spectrometer.OceanOptics'
+        spectrometer_serial: 'QEP01583' #insert here the right serial number.
+
+    """
+    _serial = ConfigOption('spectrometer_serial', missing='warn')
+
+    def on_activate(self):
+        """ Activate module.
+        """
+        self.integration_time = 1e4
```

Comment:
Hard-coding measurement parameters (especially in hardware modules) is problematic.
Isn't this initialization of the integration time unnecessary anyways? The device has a time set regardless of qudi.

Reply:
Actually, this is an important parameter and one has to be able to change it to work with the spectrometer. It should be possible to change this value in the gui. I changed it into a status variable such that it at least does not have to be set everytime.

Reply:
I was not suggesting to get rid of the possibility to change the exposure time.
I was just asking if you can read back this value from `spec`. If you can set the value with `self.spec.integration_time_micros(my_value)` there may also be a possibility to read `self.spec.integration_time_micros`. This would get rid of a possible error source and reduce redundancy by getting rid of the instance variable `self.integration_time`.
You would then still be able to set and get the integration time by calling `setExposure` and `getExposure`.

Reply:
I looked into the code of the Seabreeze module: https://github.com/ap--/python-seabreeze/blob/master/seabreeze/spectrometers.py
Interestingly it does not have a way to query the integration time. Therefore I think Jochens implementation is valid.

Reply:
Ok, cool, that's all I wanted to know. In that case the current solution is the best option.